overlord/snapstate/snapstate.go

```diff
@@ -729,6 +730,18 @@ var ValidateRefreshes func(st *state.State, refreshes []*snap.Info, ignoreValida
 // store says is updateable. If the list is empty, update everything.
 // Note that the state must be locked by the caller.
 func UpdateMany(ctx context.Context, st *state.State, names []string, userID int, flags *Flags) ([]string, []*state.TaskSet, error) {
+	return updateManyFromChange(ctx, st, "", names, userID, nil, flags)
+}
+
+// updateFilter is the type of function that can be passed to
+// updateManyFromChange so it filters the updates.
+//
+// If the filter returns true, the update for that snap proceeds. If
+// it returns false, the snap is removed from the list of updates to
+// consider.
+type updateFilter func(*snap.Info, *SnapState) bool
+
+func updateManyFromChange(ctx context.Context, st *state.State, fromChange string, names []string, userID int, filter updateFilter, flags *Flags) ([]string, []*state.TaskSet, error) {
```

Comment:
s/FromChange/Filtered/ ? also fromChange should again preferably be the last argument